src/Microsoft.AspNet.Razor/CodeGenerators/CSharpCodeWriter.cs

```diff
@@ -445,18 +458,26 @@ private void WriteVerbatimStringLiteral(string literal)
         {
             Write("@\"");
 
-            foreach (char c in literal)
+            // We need to find the index of each '"' (double-quote) to escape it.
+            var start = 0;
+            int end;
+            while ((end = literal.IndexOf('\"', start)) > -1)
             {
-                if (c == '\"')
+                if (end > start)
                 {
-                    Write("\"\"");
-                }
-                else
-                {
-                    Write(c.ToString());
```

Comment:
This addresses [my comment](https://github.com/aspnet/Razor/issues/635#issuecomment-168457369) on #635.